be/src/exprs/map_functions.cpp

```diff
@@ -124,6 +124,76 @@ StatusOr<ColumnPtr> MapFunctions::map_from_arrays(FunctionContext* context, cons
     }
 }
 
+
+StatusOr<ColumnPtr> MapFunctions::map_from_entries(FunctionContext* context, const Columns& columns) {
+    DCHECK_EQ(1, columns.size());
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    auto keys_column = ColumnHelper::unpack_and_duplicate_const_column(columns[0]->size(), columns[0]);
+    NullColumn* keys_null = nullptr;
+    ArrayColumn* keys_data = nullptr;
+    if (keys_column->is_nullable()) {
+        auto keys = down_cast<NullableColumn*>(keys_column.get());
+        keys_null = keys->null_column().get();
+        keys_data = down_cast<ArrayColumn*>(ColumnHelper::get_data_column(keys));
+    } else {
+        keys_data = down_cast<ArrayColumn*>(keys_column.get());
+    }
+
+    auto& keys_offsets = keys_data->offsets().get_data();
+
+    ColumnPtr& elements = keys_data->elements_column();
+    StructColumn* struct_column = nullptr;
+    if (elements->is_nullable()) {
+        auto keys = down_cast<NullableColumn*>(elements.get());
+        struct_column = down_cast<StructColumn*>(ColumnHelper::get_data_column(keys));
+    } else {
+        struct_column = down_cast<StructColumn*>(elements.get());
+    }
+    DCHECK_EQ(2, struct_column->fields().size());
+
+    auto num_rows = keys_data->size();
+    if (!keys_column->has_null()) {
+        for (int i = 0; i < num_rows; ++i) {
+            Datum v = keys_data->get(i);
+            const auto& items = v.get<DatumArray>();
+            for (const auto& item : items) {
+                DCHECK_EQ(2, item.get_struct().size());
+            }
+        }
+        auto copied_key_elements = struct_column->fields()[0]->clone_shared();
+        auto copied_value_elements = struct_column->fields()[1]->clone_shared();
+        auto copied_offsets = keys_data->offsets().clone_shared();
+        auto map_column = MapColumn::create(std::move(copied_key_elements), std::move(copied_value_elements),
+                                            std::static_pointer_cast<UInt32Column>(copied_offsets));
+        map_column->remove_duplicated_keys();
```

Comment:
clone_shared and remove_duplicated_keys call memcpy twice, suggest to optimized it into once.